flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTestS3ITCase.java

```diff
@@ -173,36 +175,44 @@ private void testRecursiveUploadForYarn(String scheme, String pathSuffix) throws
         }
     }
 
-    @Test
-    @RetryOnFailure(times = 3)
-    public void testRecursiveUploadForYarnS3n() throws Exception {
+    @TestTemplate
+    @RetryOnException(times = 3, exception = Exception.class)
+    public void testRecursiveUploadForYarnS3n(@TempDir File tempFolder) throws Exception {
         // skip test on Hadoop 3: https://issues.apache.org/jira/browse/HADOOP-14738
-        Assume.assumeTrue(
-                "This test is skipped for Hadoop versions above 3",
-                VersionUtil.compareVersions(System.getProperty("hadoop.version"), "3.0.0") < 0);
+        assumeTrue(
+                VersionUtil.compareVersions(System.getProperty("hadoop.version"), "3.0.0") < 0,
+                "This test is skipped for Hadoop versions above 3");
 
         try {
             Class.forName("org.apache.hadoop.fs.s3native.NativeS3FileSystem");
         } catch (ClassNotFoundException e) {
             // not in the classpath, cannot run this test
             String msg = "Skipping test because NativeS3FileSystem is not in the class path";
             log.info(msg);
-            assumeNoException(msg, e);
+            assumingThat(
+                    true,
+                    () -> {
+                        throw e;
+                    });
         }
```

Comment:
```
assumeThatThrownBy(() -> Class.forName("org.apache.hadoop.fs.s3native.NativeS3FileSystem")
    .as("Skipping test because NativeS3FileSystem is not in the class path")
    .isNull();
```

Reply:
Done.